core/evalir.ml

```diff
@@ -693,15 +693,41 @@ struct
             in apply_cont cont env (`Table ((db, params), Value.unbox_string name, unboxed_keys, row))
           | _ -> eval_error "Error evaluating table handle"
       end
-    | Query (range, e, _t) ->
+    | Query (range, policy, e, _t) ->
        let range =
          match range with
          | None -> None
          | Some (limit, offset) ->
             Some (Value.unbox_int (value env limit), Value.unbox_int (value env offset)) in
-       if Settings.get Database.shredding then
-         begin
-           if range != None then eval_error "Range is not supported for nested queries";
+       let evaluator =
+         let open QueryPolicy in
+         match policy with
+           | Flat -> `Flat
+           | Nested -> `Nested
+           | Default ->
+               if Settings.get Database.shredding then `Nested else `Flat in
+
+       let evaluate_standard () =
+         match EvalQuery.compile env (range, e) with
+           | None -> computation env cont e
```

Comment:
To fix #756 this line needs to be changed to drop the first `offset` results and take the next `limit` results, where `range = (limit,offset)`.  One way to do this could be to explicitly insert calls to Links's take and drop functions, alternatively, adjust the continuation `cont` to inspect the returned (list) value and run `Utility.take` and `Utility.drop` on it.